tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -170,6 +171,20 @@ public void Inject<TCarrier>(SpanContext context, TCarrier carrier, Action<TCarr
                         if (localSpanContext.RawTraceId == spanContext.RawTraceId)
                         {
                             localSpanContext.AdditionalW3CTraceState += spanContext.AdditionalW3CTraceState;
+
+                            if (localSpanContext.RawSpanId != spanContext.RawSpanId)
+                            {
+                                if (!string.IsNullOrEmpty(spanContext.LastParentId) && spanContext.LastParentId != W3CTraceContextPropagator.ZeroLastParent)
+                                {
+                                    localSpanContext.LastParentId = spanContext.LastParentId;
+                                }
+                                else
+                                {
+                                    localSpanContext.LastParentId = $"{localSpanContext.SpanId:x16}";
```

Comment:
When converting to/from hex strings, we should use the helper methods in `HexString` which are optimized across all .NET versions. 
```suggestion
                                    localSpanContext.LastParentId = HexString.ToHexString(localSpanContext.SpanId);
```

Reply:
Updated in: https://github.com/DataDog/dd-trace-dotnet/pull/5721/commits/42164062f5114055ccb6a5c71c20031a6438476c